modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDeploymentResult.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.deployment.recreate.steps;
+
+import java.io.Serializable;
+import java.time.OffsetDateTime;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.activiti.engine.RuntimeService;
+import org.activiti.engine.delegate.DelegateExecution;
+import org.activiti.engine.delegate.JavaDelegate;
+import org.eclipse.xpanse.modules.database.service.DeployServiceEntity;
+import org.eclipse.xpanse.modules.database.servicerecreate.ServiceRecreateEntity;
+import org.eclipse.xpanse.modules.deployment.DeployServiceEntityHandler;
+import org.eclipse.xpanse.modules.deployment.recreate.RecreateService;
+import org.eclipse.xpanse.modules.deployment.recreate.consts.RecreateConstants;
+import org.eclipse.xpanse.modules.models.service.enums.ServiceDeploymentState;
+import org.eclipse.xpanse.modules.models.workflow.recreate.enums.RecreateStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Processing class for checking deployment status after deployment callback.
+ */
+@Slf4j
+@Component
+public class ProcessRecreateDeploymentResult implements Serializable, JavaDelegate {
+
+    private final RuntimeService runtimeService;
+    private final RecreateService recreateService;
+    private final DeployServiceEntityHandler deployServiceEntityHandler;
+
+    /**
+     * Constructor for ProcessDeploymentResult bean.
+     */
+    @Autowired
+    public ProcessRecreateDeploymentResult(RuntimeService runtimeService,
+                                           RecreateService recreateService,
+                                           DeployServiceEntityHandler deployServiceEntityHandler) {
+        this.runtimeService = runtimeService;
+        this.recreateService = recreateService;
+        this.deployServiceEntityHandler = deployServiceEntityHandler;
+    }
+
+    @Override
+    public void execute(DelegateExecution execution) {
+
+        String processInstanceId = execution.getProcessInstanceId();
+        Map<String, Object> variables = runtimeService.getVariables(processInstanceId);
+        UUID originalServiceId = (UUID) variables.get(RecreateConstants.ID);
+        log.info("Recreate workflow of Instance Id : {} check deploy service status with id:{}",
+                processInstanceId, originalServiceId.toString());
```

Comment:
i have not checked this in detail. But pls ensure that the same entry in DEPLOY_SERVICE table is used to destroy and also deploy again.